From 43862e47d2b10c442b686c655fa8a67691d466a8 Mon Sep 17 00:00:00 2001
From: YukiLeafX <yukileafx@gmail.com>
Date: Fri, 15 May 2020 15:56:59 +0900
Subject: [PATCH] Don't write files with low utilization


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 7c82e18b..97859ecf 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -303,7 +303,8 @@ public final class CraftServer implements Server {
 
     private void saveCommandsConfig() {
         try {
-            commandsConfiguration.save(getCommandsConfigFile());
+            // commandsConfiguration.save(getCommandsConfigFile()); // Yukkit
+            Logger.getLogger("Yukkit").log(Level.INFO, ChatColor.AQUA + "Yukkit" + ChatColor.RESET + " >> Ignoring SAVE for " + getCommandsConfigFile()); // Yukkit
         } catch (IOException ex) {
             Logger.getLogger(CraftServer.class.getName()).log(Level.SEVERE, "Could not save " + getCommandsConfigFile(), ex);
         }
@@ -834,7 +835,8 @@ public final class CraftServer implements Server {
             stream = new FileInputStream(file);
         } catch (FileNotFoundException ex) {
             try {
-                file.createNewFile();
+                // file.createNewFile(); // Yukkit
+                Logger.getLogger("Yukkit").log(Level.INFO, ChatColor.AQUA + "Yukkit" + ChatColor.RESET + " >> Ignoring CREATE for " + file); // Yukkit
             } finally {
                 return;
             }
@@ -843,7 +845,8 @@ public final class CraftServer implements Server {
         Map<String, Map<String, Object>> perms;
 
         try {
-            perms = (Map<String, Map<String, Object>>) yaml.load(stream);
+            perms = null; // Yukkit
+            Logger.getLogger("Yukkit").log(Level.INFO, ChatColor.AQUA + "Yukkit" + ChatColor.RESET + " >> Applying empty permissions as " + file); // Yukkit
         } catch (MarkedYAMLException ex) {
             getLogger().log(Level.WARNING, "Server permissions file " + file + " is not valid YAML: " + ex.toString());
             return;
diff --git a/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java b/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
index 60a6221b..9e1a0206 100644
--- a/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
+++ b/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
@@ -37,7 +37,8 @@ public class HelpYamlReader {
 
             try {
                 if (!helpYamlFile.exists()) {
-                    helpYaml.save(helpYamlFile);
+                    // helpYaml.save(helpYamlFile); // Yukkit
+                    java.util.logging.Logger.getLogger("Yukkit").log(Level.INFO, ChatColor.AQUA + "Yukkit" + ChatColor.RESET + " >> Ignoring SAVE for " + helpYamlFile); // Yukkit
                 }
             } catch (IOException ex) {
                 server.getLogger().log(Level.SEVERE, "Could not save " + helpYamlFile, ex);
-- 
2.26.2.windows.1

