From 442363e75bfea9930d9269f043afce5a81c4cf3f Mon Sep 17 00:00:00 2001
From: YukiLeafX <yukileafx@gmail.com>
Date: Fri, 15 May 2020 20:07:07 +0900
Subject: [PATCH] Optimize files


diff --git a/src/main/java/net/minecraft/server/DedicatedServer.java b/src/main/java/net/minecraft/server/DedicatedServer.java
index 6a8292ce..2d7308e5 100644
--- a/src/main/java/net/minecraft/server/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/DedicatedServer.java
@@ -142,7 +142,7 @@ public class DedicatedServer extends MinecraftServer implements IMinecraftServer
         this.propertyManager = new PropertyManager(this.options); // CraftBukkit - CLI argument support
         this.r = new EULA(new File("eula.txt"));
         // Spigot Start
-        boolean eulaAgreed = Boolean.getBoolean( "com.mojang.eula.agree" );
+        boolean eulaAgreed = Boolean.getBoolean( "com.mojang.eula.agree" ) || Boolean.getBoolean("eula"); // Yukkit - short eula flag
         if ( eulaAgreed )
         {
             System.err.println( "You have used the Spigot command line EULA agreement flag." );
diff --git a/src/main/java/net/minecraft/server/EULA.java b/src/main/java/net/minecraft/server/EULA.java
index 3623751b..95ffa250 100644
--- a/src/main/java/net/minecraft/server/EULA.java
+++ b/src/main/java/net/minecraft/server/EULA.java
@@ -8,6 +8,9 @@ import org.apache.commons.io.IOUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.util.logging.Level; // Yukkit
+import org.bukkit.ChatColor; // Yukkit
+
 public class EULA {
 
     private static final Logger a = LogManager.getLogger();
@@ -31,7 +34,8 @@ public class EULA {
             flag = Boolean.parseBoolean(properties.getProperty("eula", "false"));
         } catch (Exception exception) {
             EULA.a.warn("Failed to load {}", file);
-            this.b();
+            java.util.logging.Logger.getLogger("Minecraft").log(Level.INFO, ChatColor.AQUA.toString() + "Yukkit" + ChatColor.RESET.toString() + " >> Ignoring SAVE for " + ChatColor.DARK_GRAY.toString() + "eula.txt" + ChatColor.RESET.toString() + "."); // Yukkit - log
+            return false; // Yukkit - don't save eula.txt when loading fails
         } finally {
             IOUtils.closeQuietly(fileinputstream);
         }
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 7c82e18b..43ec8a5c 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -302,11 +302,7 @@ public final class CraftServer implements Server {
     }
 
     private void saveCommandsConfig() {
-        try {
-            commandsConfiguration.save(getCommandsConfigFile());
-        } catch (IOException ex) {
-            Logger.getLogger(CraftServer.class.getName()).log(Level.SEVERE, "Could not save " + getCommandsConfigFile(), ex);
-        }
+        Logger.getLogger("Minecraft").log(Level.INFO, ChatColor.AQUA.toString() + "Yukkit" + ChatColor.RESET + " >> Ignoring SAVE for " + ChatColor.DARK_GRAY.toString() + getCommandsConfigFile() + ChatColor.RESET.toString() + "."); // Yukkit - log
     }
 
     public void loadPlugins() {
@@ -830,31 +826,12 @@ public final class CraftServer implements Server {
         File file = new File(configuration.getString("settings.permissions-file"));
         FileInputStream stream;
 
-        try {
-            stream = new FileInputStream(file);
-        } catch (FileNotFoundException ex) {
-            try {
-                file.createNewFile();
-            } finally {
-                return;
-            }
-        }
+        Logger.getLogger("Minecraft").log(Level.INFO, ChatColor.AQUA.toString() + "Yukkit" + ChatColor.RESET.toString() + " >> Ignoring CREATE for " + ChatColor.DARK_GRAY.toString() + file + ChatColor.RESET.toString() + "."); // Yukkit - log
 
         Map<String, Map<String, Object>> perms;
 
-        try {
-            perms = (Map<String, Map<String, Object>>) yaml.load(stream);
-        } catch (MarkedYAMLException ex) {
-            getLogger().log(Level.WARNING, "Server permissions file " + file + " is not valid YAML: " + ex.toString());
-            return;
-        } catch (Throwable ex) {
-            getLogger().log(Level.WARNING, "Server permissions file " + file + " is not valid YAML.", ex);
-            return;
-        } finally {
-            try {
-                stream.close();
-            } catch (IOException ex) {}
-        }
+        perms = null; // Yukkit - empty permissions
+        Logger.getLogger("Minecraft").log(Level.INFO, ChatColor.AQUA.toString() + "Yukkit" + ChatColor.RESET.toString() + " >> Applying empty permissions as " + ChatColor.DARK_GRAY.toString() + file + ChatColor.RESET.toString() + "."); // Yukkit - log
 
         if (perms == null) {
             getLogger().log(Level.INFO, "Server permissions file " + file + " is empty, ignoring it");
diff --git a/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java b/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
index 60a6221b..865937ba 100644
--- a/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
+++ b/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
@@ -15,6 +15,8 @@ import java.util.LinkedList;
 import java.util.List;
 import java.util.logging.Level;
 
+import java.util.logging.Logger; // Yukkit
+
 /**
  * HelpYamlReader is responsible for processing the contents of the help.yml file.
  */
@@ -35,13 +37,7 @@ public class HelpYamlReader {
             helpYaml.options().copyDefaults(true);
             helpYaml.setDefaults(defaultConfig);
 
-            try {
-                if (!helpYamlFile.exists()) {
-                    helpYaml.save(helpYamlFile);
-                }
-            } catch (IOException ex) {
-                server.getLogger().log(Level.SEVERE, "Could not save " + helpYamlFile, ex);
-            }
+            Logger.getLogger("Minecraft").log(Level.INFO, ChatColor.AQUA.toString() + "Yukkit" + ChatColor.RESET.toString() + " >> Ignoring SAVE for " + ChatColor.DARK_GRAY.toString() + helpYamlFile + ChatColor.RESET.toString() + "."); // Yukkit - log
         } catch (Exception ex) {
             server.getLogger().severe("Failed to load help.yml. Verify the yaml indentation is correct. Reverting to default help.yml.");
             helpYaml = defaultConfig;
-- 
2.26.2.windows.1

