From 3a84a1c8191a398bf92004a88ca84806e22d8bcf Mon Sep 17 00:00:00 2001
From: YukiLeafX <yukileafx@gmail.com>
Date: Sat, 16 May 2020 09:20:49 +0900
Subject: [PATCH] Optimize EULA


diff --git a/src/main/java/net/minecraft/server/DedicatedServer.java b/src/main/java/net/minecraft/server/DedicatedServer.java
index 6a8292ce4..2d7308e5d 100644
--- a/src/main/java/net/minecraft/server/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/DedicatedServer.java
@@ -142,7 +142,7 @@ public class DedicatedServer extends MinecraftServer implements IMinecraftServer
         this.propertyManager = new PropertyManager(this.options); // CraftBukkit - CLI argument support
         this.r = new EULA(new File("eula.txt"));
         // Spigot Start
-        boolean eulaAgreed = Boolean.getBoolean( "com.mojang.eula.agree" );
+        boolean eulaAgreed = Boolean.getBoolean( "com.mojang.eula.agree" ) || Boolean.getBoolean("eula"); // Yukkit - short eula flag
         if ( eulaAgreed )
         {
             System.err.println( "You have used the Spigot command line EULA agreement flag." );
diff --git a/src/main/java/net/minecraft/server/EULA.java b/src/main/java/net/minecraft/server/EULA.java
index 3623751bf..2b6673df7 100644
--- a/src/main/java/net/minecraft/server/EULA.java
+++ b/src/main/java/net/minecraft/server/EULA.java
@@ -31,7 +31,7 @@ public class EULA {
             flag = Boolean.parseBoolean(properties.getProperty("eula", "false"));
         } catch (Exception exception) {
             EULA.a.warn("Failed to load {}", file);
-            this.b();
+            return false; // Yukkit - don't save eula.txt when loading fails
         } finally {
             IOUtils.closeQuietly(fileinputstream);
         }
-- 
2.26.2.windows.1

