From b9d1194791ec1fde766109f265c1290f9b660823 Mon Sep 17 00:00:00 2001
From: YukiLeafX <yukileafx@gmail.com>
Date: Sat, 16 May 2020 09:24:29 +0900
Subject: [PATCH] Delete commands.yml


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 225d89da5..5fdf289b7 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -302,11 +302,7 @@ public final class CraftServer implements Server {
     }
 
     private void saveCommandsConfig() {
-        try {
-            commandsConfiguration.save(getCommandsConfigFile());
-        } catch (IOException ex) {
-            Logger.getLogger(CraftServer.class.getName()).log(Level.SEVERE, "Could not save " + getCommandsConfigFile(), ex);
-        }
+        // Yukkit - don't save commands.yml
     }
 
     public void loadPlugins() {
-- 
2.26.2.windows.1

