From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: YukiLeafX <yukileafx@gmail.com>
Date: Mon, 25 May 2020 21:18:00 +0900
Subject: [PATCH] Async write FileConfiguration


diff --git a/src/main/java/org/bukkit/configuration/file/FileConfiguration.java b/src/main/java/org/bukkit/configuration/file/FileConfiguration.java
index 6767d34a..7f4e9080 100644
--- a/src/main/java/org/bukkit/configuration/file/FileConfiguration.java
+++ b/src/main/java/org/bukkit/configuration/file/FileConfiguration.java
@@ -63,17 +63,7 @@ public abstract class FileConfiguration extends MemoryConfiguration {
     public void save(File file) throws IOException {
         Validate.notNull(file, "File cannot be null");
 
-        Files.createParentDirs(file);
-
-        String data = saveToString();
-
-        Writer writer = new OutputStreamWriter(new FileOutputStream(file), Charsets.UTF_8);
-
-        try {
-            writer.write(data);
-        } finally {
-            writer.close();
-        }
+        io.github.yukileafx.yukkit.AsyncConfigWriter.INSTANCE.write(file, this); // Yukkit
     }
 
     /**
-- 
2.26.2.windows.1

