From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: YukiLeafX <yukileafx@gmail.com>
Date: Thu, 28 May 2020 15:04:06 +0900
Subject: [PATCH] Gradle


diff --git a/build.gradle.kts b/build.gradle.kts
new file mode 100644
index 00000000..3f64a6bd
--- /dev/null
+++ b/build.gradle.kts
@@ -0,0 +1,48 @@
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+
+plugins {
+    java
+    id("com.github.johnrengelman.shadow") version "5.2.0"
+}
+
+repositories {
+    mavenCentral()
+    maven("https://hub.spigotmc.org/nexus/content/groups/public/")
+    maven("https://papermc.io/repo/repository/maven-public/")
+    maven("https://oss.sonatype.org/content/groups/public/")
+}
+
+dependencies {
+    // Yukkit-API
+    implementation(project(":modules:Yukkit-Core"))
+    // Paper-API
+    implementation("com.google.code.findbugs:jsr305:1.3.9")
+    compileOnly("net.sf.trove4j:trove4j:3.0.3")
+    compileOnly("co.aikar:fastutil-lite:1.0")
+    implementation("org.ow2.asm:asm:6.1.1")
+    implementation("org.ow2.asm:asm-commons:6.1.1")
+    implementation("com.mojang:authlib:1.5.25")
+    implementation("org.slf4j:slf4j-api:1.7.25")
+    // Spigot-API
+    implementation("net.md-5:bungeecord-chat:1.12-SNAPSHOT")
+    // Bukkit
+    implementation("commons-lang:commons-lang:2.6")
+    implementation("com.googlecode.json-simple:json-simple:1.1.1")
+    implementation("com.google.guava:guava:21.0")
+    implementation("com.google.code.gson:gson:2.8.0")
+    implementation("org.yaml:snakeyaml:1.19")
+    testImplementation("junit:junit:4.12")
+    testImplementation("org.hamcrest:hamcrest-library:1.3")
+}
+
+configurations {
+    create("depends")
+}
+
+artifacts {
+    add("depends", tasks["shadowJar"])
+}
+
+tasks.withType<ShadowJar> {
+    minimize()
+}
-- 
2.26.2.windows.1

