From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: YukiLeafX <yukileafx@gmail.com>
Date: Thu, 28 May 2020 03:53:46 +0900
Subject: [PATCH] Add a system property to disable recipe warnings


diff --git a/src/main/java/org/bukkit/inventory/ShapedRecipe.java b/src/main/java/org/bukkit/inventory/ShapedRecipe.java
index d7fa6d28f..849764e96 100644
--- a/src/main/java/org/bukkit/inventory/ShapedRecipe.java
+++ b/src/main/java/org/bukkit/inventory/ShapedRecipe.java
@@ -23,7 +23,9 @@ public class ShapedRecipe implements Recipe, Keyed {
     @Deprecated
     public ShapedRecipe(ItemStack result) {
         this.key = NamespacedKey.randomKey();
-        new Throwable("Warning: A plugin is creating a recipe using a Deprecated method. This will cause you to receive warnings stating 'Tried to load unrecognized recipe: bukkit:<ID>'. Please ask the author to give their recipe a static key using NamespacedKey.").printStackTrace();
+        if (!Boolean.getBoolean("yukkit.disable-recipe-warnings")) {
+            new Throwable("Warning: A plugin is creating a recipe using a Deprecated method. This will cause you to receive warnings stating 'Tried to load unrecognized recipe: bukkit:<ID>'. Please ask the author to give their recipe a static key using NamespacedKey.").printStackTrace();
+        }
         this.output = new ItemStack(result);
     }
 
diff --git a/src/main/java/org/bukkit/inventory/ShapelessRecipe.java b/src/main/java/org/bukkit/inventory/ShapelessRecipe.java
index f2dd46f27..c8c3b607c 100644
--- a/src/main/java/org/bukkit/inventory/ShapelessRecipe.java
+++ b/src/main/java/org/bukkit/inventory/ShapelessRecipe.java
@@ -23,7 +23,9 @@ public class ShapelessRecipe implements Recipe, Keyed {
     @Deprecated
     public ShapelessRecipe(ItemStack result) {
         this.key = NamespacedKey.randomKey();
-        new Throwable("Warning: A plugin is creating a recipe using a Deprecated method. This will cause you to receive warnings stating 'Tried to load unrecognized recipe: bukkit:<ID>'. Please ask the author to give their recipe a static key using NamespacedKey.").printStackTrace();
+        if (!Boolean.getBoolean("yukkit.disable-recipe-warnings")) {
+            new Throwable("Warning: A plugin is creating a recipe using a Deprecated method. This will cause you to receive warnings stating 'Tried to load unrecognized recipe: bukkit:<ID>'. Please ask the author to give their recipe a static key using NamespacedKey.").printStackTrace();
+        }
         this.output = new ItemStack(result);
     }
 
-- 
2.26.2.windows.1

