From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: YukiLeafX <yukileafx@gmail.com>
Date: Tue, 26 May 2020 01:26:54 +0900
Subject: [PATCH] Add JVM flags to java commands


diff --git a/scripts/decompile.sh b/scripts/decompile.sh
index 032abdb7..f01ae217 100755
--- a/scripts/decompile.sh
+++ b/scripts/decompile.sh
@@ -24,7 +24,7 @@ fi
 echo "Decompiling classes..."
 if [ ! -d "$decompiledir/net/minecraft/server" ]; then
     cd "$basedir"
-    java -jar "$workdir/BuildData/bin/fernflower.jar" -dgs=1 -hdc=0 -asc=1 -udv=0 "$classdir" "$decompiledir"
+    java -Xms2G -Xmx2G -XX:+UseConcMarkSweepGC -XX:+DisableExplicitGC -jar "$workdir/BuildData/bin/fernflower.jar" -dgs=1 -hdc=0 -asc=1 -udv=0 "$classdir" "$decompiledir"
     if [ "$?" != "0" ]; then
         echo "Failed to decompile classes."
         exit 1
diff --git a/scripts/remap.sh b/scripts/remap.sh
index c64bcca8..96f5af37 100755
--- a/scripts/remap.sh
+++ b/scripts/remap.sh
@@ -43,7 +43,7 @@ fi
 
 echo "Applying class mappings..."
 if [ ! -f "$jarpath-cl.jar" ]; then
-    java -jar "$workdir/BuildData/bin/SpecialSource-2.jar" map -i "$jarpath.jar" -m "$classmappings" -o "$jarpath-cl.jar" 1>/dev/null
+    java -Xms2G -Xmx2G -XX:+UseConcMarkSweepGC -XX:+DisableExplicitGC -jar "$workdir/BuildData/bin/SpecialSource-2.jar" map -i "$jarpath.jar" -m "$classmappings" -o "$jarpath-cl.jar" 1>/dev/null
     if [ "$?" != "0" ]; then
         echo "Failed to apply class mappings."
         exit 1
@@ -52,7 +52,7 @@ fi
 
 echo "Applying member mappings..."
 if [ ! -f "$jarpath-m.jar" ]; then
-    java -jar "$workdir/BuildData/bin/SpecialSource-2.jar" map -i "$jarpath-cl.jar" -m "$membermappings" -o "$jarpath-m.jar" 1>/dev/null
+    java -Xms2G -Xmx2G -XX:+UseConcMarkSweepGC -XX:+DisableExplicitGC -jar "$workdir/BuildData/bin/SpecialSource-2.jar" map -i "$jarpath-cl.jar" -m "$membermappings" -o "$jarpath-m.jar" 1>/dev/null
     if [ "$?" != "0" ]; then
         echo "Failed to apply member mappings."
         exit 1
@@ -61,7 +61,7 @@ fi
 
 echo "Creating remapped jar..."
 if [ ! -f "$jarpath-mapped.jar" ]; then
-    java -jar "$workdir/BuildData/bin/SpecialSource.jar" --kill-lvt -i "$jarpath-m.jar" --access-transformer "$accesstransforms" -m "$packagemappings" -o "$jarpath-mapped.jar" 1>/dev/null
+    java -Xms2G -Xmx2G -XX:+UseConcMarkSweepGC -XX:+DisableExplicitGC -jar "$workdir/BuildData/bin/SpecialSource.jar" --kill-lvt -i "$jarpath-m.jar" --access-transformer "$accesstransforms" -m "$packagemappings" -o "$jarpath-mapped.jar" 1>/dev/null
     if [ "$?" != "0" ]; then
         echo "Failed to create remapped jar."
         exit 1
-- 
2.26.2.windows.1

