From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: YukiLeafX <yukileafx@gmail.com>
Date: Tue, 2 Jun 2020 14:42:42 +0900
Subject: [PATCH] Get API version


diff --git a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
index 674096cab..f6af8f050 100644
--- a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
@@ -11,7 +11,7 @@ public final class Versioning {
     public static String getBukkitVersion() {
         String result = "Unknown-Version";
 
-        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/com.destroystokyo.paper/paper-api/pom.properties");
+        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("version.properties");
         Properties properties = new Properties();
 
         if (stream != null) {
-- 
2.26.2.windows.1

