From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: YukiLeafX <yukileafx@gmail.com>
Date: Sat, 16 May 2020 09:21:31 +0900
Subject: [PATCH] Don't save underused files


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 2e4b8fe64..71790216e 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -303,8 +303,8 @@ public final class CraftServer implements Server {
 
     private void saveCommandsConfig() {
         try {
-            commandsConfiguration.save(getCommandsConfigFile());
-        } catch (IOException ex) {
+            // Yukkit - don't save commands.yml
+        } catch (Exception ex) { // Yukkit
             Logger.getLogger(CraftServer.class.getName()).log(Level.SEVERE, "Could not save " + getCommandsConfigFile(), ex);
         }
     }
@@ -834,7 +834,7 @@ public final class CraftServer implements Server {
             stream = new FileInputStream(file);
         } catch (FileNotFoundException ex) {
             try {
-                file.createNewFile();
+                // Yukkit - don't save permissions.yml
             } finally {
                 return;
             }
diff --git a/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java b/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
index 60a6221b4..962d45fa1 100644
--- a/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
+++ b/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
@@ -37,9 +37,9 @@ public class HelpYamlReader {
 
             try {
                 if (!helpYamlFile.exists()) {
-                    helpYaml.save(helpYamlFile);
+                    // Yukkit - don't save help.yml
                 }
-            } catch (IOException ex) {
+            } catch (Exception ex) { // Yukkit
                 server.getLogger().log(Level.SEVERE, "Could not save " + helpYamlFile, ex);
             }
         } catch (Exception ex) {
-- 
2.26.2.windows.1

