From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: YukiLeafX <yukileafx@gmail.com>
Date: Wed, 3 Jun 2020 23:58:32 +0900
Subject: [PATCH] Improve CraftAsyncScheduler management executor performance


diff --git a/src/main/java/org/bukkit/craftbukkit/scheduler/CraftAsyncScheduler.java b/src/main/java/org/bukkit/craftbukkit/scheduler/CraftAsyncScheduler.java
index eaf869287..ea37ad240 100644
--- a/src/main/java/org/bukkit/craftbukkit/scheduler/CraftAsyncScheduler.java
+++ b/src/main/java/org/bukkit/craftbukkit/scheduler/CraftAsyncScheduler.java
@@ -41,8 +41,7 @@ public class CraftAsyncScheduler extends CraftScheduler {
     private final ThreadPoolExecutor executor = new ThreadPoolExecutor(
             4, Integer.MAX_VALUE,30L, TimeUnit.SECONDS, new SynchronousQueue<>(),
             new ThreadFactoryBuilder().setNameFormat("Craft Scheduler Thread - %1$d").build());
-    private final Executor management = Executors.newSingleThreadExecutor(new ThreadFactoryBuilder()
-            .setNameFormat("Craft Async Scheduler Management Thread").build());
+    private final ThreadPoolExecutor management = io.github.yukileafx.yukkit.Yukkit.INSTANCE.newSingleExecutor("Craft Async Scheduler Management Thread"); // Yukkit
     private final List<CraftTask> temp = new ArrayList<>();
 
     CraftAsyncScheduler() {
-- 
2.26.2.windows.1

