From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: YukiLeafX <yukileafx@gmail.com>
Date: Sat, 16 May 2020 18:45:49 +0900
Subject: [PATCH] Apply compact settings to server


diff --git a/src/main/java/com/destroystokyo/paper/PaperConfig.java b/src/main/java/com/destroystokyo/paper/PaperConfig.java
index cc0360660..289495cb2 100644
--- a/src/main/java/com/destroystokyo/paper/PaperConfig.java
+++ b/src/main/java/com/destroystokyo/paper/PaperConfig.java
@@ -57,6 +57,7 @@ public class PaperConfig {
             Bukkit.getLogger().log(Level.SEVERE, "Could not load paper.yml, please correct your syntax errors", ex);
             throw Throwables.propagate(ex);
         }
+        io.github.yukileafx.YukkitConfig.SETTINGS.get("paper-yml").apply("paper.yml", config); // Yukkit
         config.options().header(HEADER);
         config.options().copyDefaults(true);
         verbose = getBoolean("verbose", false);
diff --git a/src/main/java/io/github/yukileafx/YukkitConfig.java b/src/main/java/io/github/yukileafx/YukkitConfig.java
new file mode 100644
index 000000000..62a995725
--- /dev/null
+++ b/src/main/java/io/github/yukileafx/YukkitConfig.java
@@ -0,0 +1,121 @@
+package io.github.yukileafx;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Properties;
+import java.util.TreeMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Resources;
+
+import org.bukkit.ChatColor;
+import org.bukkit.configuration.ConfigurationSection;
+import org.bukkit.configuration.InvalidConfigurationException;
+import org.bukkit.configuration.file.YamlConfiguration;
+
+public class YukkitConfig {
+
+    private static final Logger LOGGER = Logger.getLogger("Minecraft");
+
+    public static final Map<String, YukkitConfig> SETTINGS = ImmutableMap.of( //
+            "server-properties", new YukkitConfig(), //
+            "bukkit-yml", new YukkitConfig(), //
+            "spigot-yml", new YukkitConfig(), //
+            "paper-yml", new YukkitConfig());
+
+    private static void copyResource(String name, Path path) {
+        URL url = Resources.getResource(name);
+        try {
+            boolean nf = !Files.isRegularFile(path);
+            boolean df = !nf && !Arrays.equals(Resources.toByteArray(url), Files.readAllBytes(path));
+            if (nf || df) {
+                String causedBy = nf ? "file not found" : df ? "content different" : "";
+                LOGGER.log(Level.INFO, "{0}Updating{1} | {2} -> {3} (Triggered by {4})", //
+                        new Object[] { ChatColor.AQUA, ChatColor.RESET, name, path, causedBy });
+                Resources.copy(url, Files.newOutputStream(path));
+            }
+        } catch (IOException e) {
+            LOGGER.log(Level.SEVERE, "Unable to copy resource " + name + " -> " + path, e);
+        }
+    }
+
+    public static void init() {
+        File dir = new File("settings");
+        dir.mkdirs();
+
+        Path yukkit = dir.toPath().resolve("00-yukkit.yml");
+        Path bungee = dir.toPath().resolve("01-yukkit-behind-bungee.yml");
+
+        copyResource("settings/yukkit.yml", yukkit);
+
+        if (Boolean.getBoolean("bungee")) {
+            copyResource("settings/yukkit-behind-bungee.yml", bungee);
+        } else {
+            try {
+                Files.deleteIfExists(bungee);
+            } catch (IOException e) {
+                LOGGER.log(Level.SEVERE, "Unable to delete " + bungee, e);
+            }
+        }
+
+        try {
+            Files.list(dir.toPath()).forEach(path -> {
+                YamlConfiguration yaml = new YamlConfiguration();
+                try {
+                    yaml.load(path.toFile());
+                } catch (IOException | InvalidConfigurationException e) {
+                    LOGGER.log(Level.SEVERE, "Settings file " + path + " is invalid syntax. Skipping...", e);
+                }
+                yaml.getKeys(false).forEach(id -> {
+                    ConfigurationSection section = yaml.getConfigurationSection(id);
+                    if (section == null || !SETTINGS.containsKey(id)) {
+                        LOGGER.log(Level.WARNING, "Invalid section " + id + " in " + path + ". Ignoring...");
+                        return;
+                    }
+                    section.getValues(true).entrySet().stream() //
+                            .filter(entry -> entry.getKey() != null)
+                            .filter(entry -> !(entry.getValue() instanceof ConfigurationSection))
+                            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))
+                            .forEach(SETTINGS.get(id).contents::put);
+                });
+            });
+        } catch (IOException e) {
+            LOGGER.log(Level.SEVERE, "Unable to read configuration files in " + dir, e);
+        }
+    }
+
+    private final Map<String, Object> defaults = new TreeMap<>();
+
+    private final Map<String, Object> contents = new TreeMap<>();
+
+    protected YukkitConfig() {
+    }
+
+    public void addDefault(String key, Object value) {
+        defaults.putIfAbsent(key, value);
+    }
+
+    public void apply(String name, Properties prop) {
+        contents.forEach((key, value) -> {
+            LOGGER.log(Level.INFO, "{0}Applying{1} | {2} @ {3} = {4}", //
+                    new Object[] { ChatColor.GREEN, ChatColor.RESET, name, key, value });
+            prop.setProperty(key, value + "");
+        });
+    }
+
+    public void apply(String name, YamlConfiguration yaml) {
+        contents.forEach((key, value) -> {
+            LOGGER.log(Level.INFO, "{0}Applying{1} | {2} @ {3} = {4}", //
+                    new Object[] { ChatColor.GREEN, ChatColor.RESET, name, key, value });
+            yaml.set(key, value);
+        });
+    }
+}
diff --git a/src/main/java/net/minecraft/server/DedicatedServer.java b/src/main/java/net/minecraft/server/DedicatedServer.java
index 2d7308e5d..4f503f09d 100644
--- a/src/main/java/net/minecraft/server/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/DedicatedServer.java
@@ -155,6 +155,11 @@ public class DedicatedServer extends MinecraftServer implements IMinecraftServer
             this.r.b();
             return false;
         } else {
+            // Yukkit start
+            io.github.yukileafx.YukkitConfig.init();
+            io.github.yukileafx.YukkitConfig.SETTINGS.get("server-properties").apply("server.properties", propertyManager.properties);
+            // Yukkit end
+
             if (this.R()) {
                 this.c("127.0.0.1");
             } else {
diff --git a/src/main/java/net/minecraft/server/PropertyManager.java b/src/main/java/net/minecraft/server/PropertyManager.java
index 8288f656f..79880e1fe 100644
--- a/src/main/java/net/minecraft/server/PropertyManager.java
+++ b/src/main/java/net/minecraft/server/PropertyManager.java
@@ -99,6 +99,8 @@ public class PropertyManager {
     }
 
     public String getString(String s, String s1) {
+        io.github.yukileafx.YukkitConfig.SETTINGS.get("server-properties").addDefault(s, s1); // Yukkit
+
         if (!this.properties.containsKey(s)) {
             this.properties.setProperty(s, s1);
             this.savePropertiesFile();
@@ -109,6 +111,8 @@ public class PropertyManager {
     }
 
     public int getInt(String s, int i) {
+        io.github.yukileafx.YukkitConfig.SETTINGS.get("server-properties").addDefault(s, i); // Yukkit
+
         try {
             return getOverride(s, Integer.parseInt(this.getString(s, "" + i))); // CraftBukkit
         } catch (Exception exception) {
@@ -119,6 +123,8 @@ public class PropertyManager {
     }
 
     public long getLong(String s, long i) {
+        io.github.yukileafx.YukkitConfig.SETTINGS.get("server-properties").addDefault(s, i); // Yukkit
+
         try {
             return getOverride(s, Long.parseLong(this.getString(s, "" + i))); // CraftBukkit
         } catch (Exception exception) {
@@ -129,6 +135,8 @@ public class PropertyManager {
     }
 
     public boolean getBoolean(String s, boolean flag) {
+        io.github.yukileafx.YukkitConfig.SETTINGS.get("server-properties").addDefault(s, flag); // Yukkit
+
         try {
             return getOverride(s, Boolean.parseBoolean(this.getString(s, "" + flag))); //CraftBukkit
         } catch (Exception exception) {
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index aa19507de..138700d61 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -217,6 +217,7 @@ public final class CraftServer implements Server {
         }
 
         configuration = YamlConfiguration.loadConfiguration(getConfigFile());
+        io.github.yukileafx.YukkitConfig.SETTINGS.get("bukkit-yml").apply("bukkit.yml", configuration); // Yukkit
         configuration.options().copyDefaults(true);
         configuration.setDefaults(YamlConfiguration.loadConfiguration(new InputStreamReader(getClass().getClassLoader().getResourceAsStream("configurations/bukkit.yml"), Charsets.UTF_8)));
         ConfigurationSection legacyAlias = null;
diff --git a/src/main/java/org/spigotmc/SpigotConfig.java b/src/main/java/org/spigotmc/SpigotConfig.java
index c790f1642..8b490a9f7 100644
--- a/src/main/java/org/spigotmc/SpigotConfig.java
+++ b/src/main/java/org/spigotmc/SpigotConfig.java
@@ -64,6 +64,7 @@ public class SpigotConfig
             Bukkit.getLogger().log( Level.SEVERE, "Could not load spigot.yml, please correct your syntax errors", ex );
             throw Throwables.propagate( ex );
         }
+        io.github.yukileafx.YukkitConfig.SETTINGS.get("spigot-yml").apply("spigot.yml", config); // Yukkit
 
         config.options().header( HEADER );
         config.options().copyDefaults( true );
-- 
2.26.2.windows.1

